lib/Sema/DerivedConformanceDistributedActor.cpp

```diff
@@ -23,14 +23,647 @@
+                   ->removeArgumentLabels(1));
+
+  llvm::SmallString<64> buffer;
+  StringRef fullClassName = ctx.AllocateCopy(
+      (classDecl->getModuleContext()->getName().str() +
+       "." +
+       classDecl->getName().str()).toStringRef(buffer));
+
+  auto *className = new (ctx) StringLiteralExpr(fullClassName, loc,
+      /*Implicit=*/true);
+  className->setBuiltinInitializer(staticStringInit);
+  assert(isa<ConstructorDecl>(className->getBuiltinInitializer().getDecl()));
+  className->setType(staticStringType);
+
+  auto *initName = new (ctx) MagicIdentifierLiteralExpr(
+      MagicIdentifierLiteralExpr::Function, loc, /*Implicit=*/true);
+  initName->setType(staticStringType);
+  initName->setBuiltinInitializer(staticStringInit);
+
+  auto *file = new (ctx) MagicIdentifierLiteralExpr(
+      MagicIdentifierLiteralExpr::FileID, loc, /*Implicit=*/true);
+  file->setType(staticStringType);
+  file->setBuiltinInitializer(staticStringInit);
+
+  auto *line = new (ctx) MagicIdentifierLiteralExpr(
+      MagicIdentifierLiteralExpr::Line, loc, /*Implicit=*/true);
+  line->setType(uintType);
+  line->setBuiltinInitializer(uintInit);
+
+  auto *column = new (ctx) MagicIdentifierLiteralExpr(
+      MagicIdentifierLiteralExpr::Column, loc, /*Implicit=*/true);
+  column->setType(uintType);
+  column->setBuiltinInitializer(uintInit);
+
+  auto *call = CallExpr::createImplicit(
+      ctx, ref, { className, initName, file, line, column }, {});
+  call->setType(ctx.getNeverType());
+  call->setThrows(false);
+
+  SmallVector<ASTNode, 2> stmts;
+  stmts.push_back(call);
+  stmts.push_back(new (ctx) ReturnStmt(SourceLoc(), /*Result=*/nullptr));
+  return { BraceStmt::create(ctx, SourceLoc(), stmts, SourceLoc(),
+      /*implicit=*/true),
+      /*isTypeChecked=*/true };
+}
+
+// ==== ------------------------------------------------------------------------
+
+// TODO: deduplicate with 'declareDerivedProperty' from DerivedConformance...
+// FIXME: !!!!!!!!!!!!!!!!!!!!!
+std::pair<VarDecl *, PatternBindingDecl *>
+createStoredProperty(ValueDecl *parent, DeclContext *parentDC, ASTContext &ctx,
+                     VarDecl::Introducer introducer, Identifier name,
+                     Type propertyInterfaceType, Type propertyContextType,
+                     bool isStatic, bool isFinal,
+                     llvm::Optional<AccessLevel> accessLevel = llvm::None);
+
+
+// TODO: similar to getVarNameForCoding, so maybe move it onto VarDecl
+static Identifier getVarName(VarDecl *var) {
+  if (auto originalVar = var->getOriginalWrappedProperty())
+    return originalVar->getName();
+
+  return var->getName();
+}
+
+
+/******************************************************************************/
+/*********************************** TYPES ************************************/
+/******************************************************************************/
+
+static Type
+getBoundPersonalityStorageType(ASTContext &C,
```

Comment:
^^^ This is what is the bigger issue right now; I can't figure out how to make this type pass the match checking so that it is equal to `DistributedActorStorage<Self.DistributedActorLocalStorage>` that the requirement expects:
`(var_decl "storage" type='DistributedActorStorage<Self.DistributedActorLocalStorage>' interface type='DistributedActorStorage<Self.DistributedActorLocalStorage>' access=public readImpl=getter immutable`

We end up synthesizing:

`(var_decl implicit "storage" type='DistributedActorStorage<Person.DistributedActorLocalStorage>' interface type='DistributedActorStorage<Person.DistributedActorLocalStorage>' access=internal readImpl=stored writeImpl=stored readWriteImpl=stored)`


note the **Person.**DistributedActorLocalStorage which is the issue. For some reason it does not understand that that struct directly implements that associated type from the protocol 🤔 

--- 

more context:
I am 100% the error in synthesis results from this type not being "a match" to the protocol requirement. The other variable things definitely work and are the same. I also checked if we hardcode this to some non generic thing if it works fine, it does.

Some notes:

```
[/Users/ktoso/code/swift-project-distributed/swift/lib/Sema/TypeCheckProtocol.cpp:4188] (resolveWitnessViaDerivation) derive, match KIND: 0 // exact match

[/Users/ktoso/code/swift-project-distributed/swift/lib/Sema/TypeCheckProtocol.cpp:4189] (resolveWitnessViaDerivation) derive, requirement:
(var_decl "actorAddress" type='ActorAddress' interface type='ActorAddress' access=public readImpl=getter immutable
  (accessor_decl 'anonname=0x7fbd65028c90' access=public get_for=actorAddress
    (parameter "self")
    (parameter_list)))
[/Users/ktoso/code/swift-project-distributed/swift/lib/Sema/TypeCheckProtocol.cpp:4191] (resolveWitnessViaDerivation) derive, DERIVED:
(var_decl implicit "actorAddress" type='ActorAddress' interface type='ActorAddress' access=internal let readImpl=stored immutable
  (accessor_decl implicit 'anonname=0x7fbd5480f020' get_for=actorAddress
    (parameter "self")
    (parameter_list)))
```


—

```
[/Users/ktoso/code/swift-project-distributed/swift/lib/Sema/TypeCheckProtocol.cpp:4188] (resolveWitnessViaDerivation) derive, match KIND: 6 // The types conflict - MatchKind::TypeConflict

[/Users/ktoso/code/swift-project-distributed/swift/lib/Sema/TypeCheckProtocol.cpp:4189] (resolveWitnessViaDerivation) derive, requirement:
(var_decl "storage" type='DistributedActorStorage<Self.DistributedActorLocalStorage>' interface type='DistributedActorStorage<Self.DistributedActorLocalStorage>' access=public readImpl=getter immutable
  (accessor_decl 'anonname=0x7fbd650291b0' access=public get_for=storage
    (parameter "self")
    (parameter_list)))
[/Users/ktoso/code/swift-project-distributed/swift/lib/Sema/TypeCheckProtocol.cpp:4191] (resolveWitnessViaDerivation) derive, DERIVED:
(var_decl implicit "storage" type='DistributedActorStorage<Person.DistributedActorLocalStorage>' interface type='DistributedActorStorage<Person.DistributedActorLocalStorage>' access=internal readImpl=stored writeImpl=stored readWriteImpl=stored)
/Users/ktoso/code/swift-project-distributed/swift/test/Concurrency/Runtime/distributed_actor_init_minimum.swift:8:19: error: type 'Person' does not conform to protocol 'DistributedActor'
distributed actor Person {
                  ^
/Users/ktoso/code/swift-project-distributed/swift/test/Concurrency/Runtime/distributed_actor_init_minimum.swift:8:19: error: protocol 'DistributedActor' is broken; cannot derive conformance for type 'Person'. Match kind: The types conflict.
distributed actor Person {
                  ^
```

—— 

I am 100% confident it is about the generic type there, and not other issues.
See this experiment with requiring and setting a String in there, which does not have the typealias / synthesized struct dance involved:

```
[/Users/ktoso/code/swift-project-distributed/swift/lib/Sema/TypeCheckProtocol.cpp:4188] (resolveWitnessViaDerivation) derive, match KIND: 0 // OK!
[/Users/ktoso/code/swift-project-distributed/swift/lib/Sema/TypeCheckProtocol.cpp:4189] (resolveWitnessViaDerivation) derive, requirement:
(var_decl "storage" type='DistributedActorStorage<String>' interface type='DistributedActorStorage<String>' access=public readImpl=getter writeImpl=setter readWriteImpl=modify_coroutine
  (accessor_decl 'anonname=0x7fc79b042f08' access=public get_for=storage
    (parameter "self")
    (parameter_list))
  (accessor_decl 'anonname=0x7fc79b043010' access=public set_for=storage
    (parameter "self")
    (parameter_list
      (parameter "newValue" type='DistributedActorStorage<String>' interface type='DistributedActorStorage<String>')))
  (accessor_decl implicit 'anonname=0x7fc79b0431e0' access=public _modify_for=storage
    (parameter "self")
    (parameter_list)))
[/Users/ktoso/code/swift-project-distributed/swift/lib/Sema/TypeCheckProtocol.cpp:4191] (resolveWitnessViaDerivation) derive, DERIVED:
(var_decl implicit "storage" type='DistributedActorStorage<String>' interface type='DistributedActorStorage<String>' access=internal readImpl=stored writeImpl=stored readWriteImpl=stored
  (accessor_decl implicit 'anonname=0x7fc7ac00a090' get_for=storage
    (parameter "self")
    (parameter_list)))
```

Reply:
To reproduce you can: `ninja -C ../build/Ninja-RelWithDebInfoAssert/swift-macosx-x86_64 && ../llvm-project/llvm/utils/lit/lit.py -s  -a ../build/Ninja-RelWithDebInfoAssert/swift-macosx-x86_64/test-macosx-x86_64 --filter=distributed_actor_init_minimum`